### ISSUE TITLE
mainline `cabal-edit`

### ISSUE THREAD

**Issue by alok** [haskell/cabal#7337]
the idea is to grow cabal to add some subcommands with sensible defaults. i'd be happy to implement if someone points me the right way and there's interest.

**Comment by emilypi** [comment]
I think this is a wonderful idea, and would dovetail nicely with the `cabal gen-bounds` work, for which we wanted to add a write-in-place option that required a slightly better parser. If it's already done in this package, I see no reason to not include this. CC @fgaz @mightybyte.

**Comment by fgaz** [comment]
Is cabal-edit this project https://github.com/sdiehl/cabal-edit?

Sounds like an useful addition, but it'd have to overcome the [limitations](https://github.com/sdiehl/cabal-edit#limitations) stated in the readme first, otherwise editing would not work on a good chunk of packages.
Another point to think about is how projects could affect the commands: eg. maybe `add` should check for compatibility with the rest of the project first.

**Comment by cartazio** [comment]
a related idea i had (which maybe already exists?)
would be a cabal-explain : eg explain / list what the current named compoents are (eg libraries/test/executables), i dont know if theres actually a way to list those now? @fgaz ?

**Comment by gbaz** [comment]
Here's the main stuff that needs to be resolved for a proper roundtrip one way or another as I understand it: https://github.com/haskell/cabal/issues/5306

**Comment by alok** [comment]
@gbaz looks like comment preserving exactprint is the blocker in #5306? where should i look to find that in the codebase?

**Comment by Mikolaj** [comment]
@alok: exact-print is not done yet, though we'd love to have it and there's a related project: https://github.com/haskell/cabal/projects/9

**Comment by gbaz** [comment]
cabal edit should subsume this old ticket (--save is sort of like --add here). https://github.com/haskell/cabal/issues/1961
